src/docker/manifest.nim

```diff
@@ -351,3 +320,94 @@ proc fetchImageManifest*(name: DockerImage,
       $platform & " != " & "" & $manifest.platform
     )
   return manifest
+
+proc findSibling(self: DockerManifest, reference = "attestation-manifest"): DockerManifest =
+  if self.kind != DockerManifestType.image:
+    raise newException(ValueError, "Can only lookup sibling for image manifest")
+  if self.list == nil:
+    raise newException(ValueError, "Need reference to list manifest to lookup sibling")
+  for i in self.list.manifests:
+    if i.annotations == nil:
+      continue
+    if (
+      i.annotations{"vnd.docker.reference.type"}.getStr().toLower() == reference.toLower() and
+      i.annotations{"vnd.docker.reference.digest"}.getStr().toLower() == self.nameRef.imageRef.toLower()
+    ):
+      return i.fetch()
+  raise newException(KeyError, "Could not find sibling image of reference type: " & reference)
+
+proc findInTotoLayer(self: DockerManifest, predicate: string): DockerManifest =
+  if self.kind != DockerManifestType.image:
+    raise newException(ValueError, "Can only lookup layers in image manifest")
+  for l in self.layers:
+    if l.annotations == nil:
+      continue
+    if l.annotations{"in-toto.io/predicate-type"}.getStr().toLower().startsWith(predicate.toLower()):
+      return l
+  raise newException(KeyError, "Could not find in-toto layer for predicate: " & predicate)
+
+proc fetchProvenance*(name: DockerImage, platform: DockerPlatform): JsonNode =
+  # https://docs.docker.com/reference/cli/docker/buildx/imagetools/inspect/
+  try:
+    trace("docker: looking up provenance for: " & $name)
+    let layer = name.fetchImageManifest(platform).findSibling().findInTotoLayer("https://slsa.dev/provenance/")
+    result = layer.nameRef.layerGetJson(accept = layer.mediaType).json{"predicate"}
+    trace("docker: in registry found provenance for: " & $name)
+  except RegistryResponseError, KeyError:
+    trace("docker: " & getCurrentExceptionMsg())
+    raise
+  except:
+    error("docker: " & getCurrentExceptionMsg())
+    dumpExOnDebug()
+    # https://github.com/docker/buildx/releases/tag/v0.13.0
+    if getBuildXVersion() < parseVersion("0.13"):
+      raise newException(ValueError, "buildx 0.13 is required to collect provenance via imagetools")
+    try:
+      # for single-platform manifests, there is only a single provenance
+      return requestManifestJson(
+        name,
+        flags    = @["--format", "{{json .Provenance.SLSA}}"],
+        fallback = false,
+      ).json
+    except:
+      # for multi-platform we have to filter on the platform
+      # note index only supports a few keys so have to manually
+      # take SLSA key
+      return requestManifestJson(
+        name,
+        flags    = @["--format", "{{json (index .Provenance \"" & $platform & "\")}}"],
+        fallback = false,
+      ).json{"SLSA"}
+
+proc fetchSBOM*(name: DockerImage, platform: DockerPlatform): JsonNode =
```

Comment:
nit: `fetchSBOM` -> `fetchSbom`. Would align with Nim's std lib style (e.g. `JsonNode`)

Reply:
SBOM is in itself abbreviation. most cases we use `SBOM`:

```
➜ rg 'SBOM' | wc -l
73
➜ rg 'Sbom' | wc -l
2
```

we should prolly change those 2 to match `SBOM` everywhere else